src/app/machines/views/MachineList/MachineList.test.tsx

```diff
@@ -229,192 +224,137 @@ describe("MachineList", () => {
-              hiddenGroups={[]}
-              searchFilter=""
-              setHiddenGroups={jest.fn()}
-            />
-          </CompatRouter>
-        </MemoryRouter>
-      </Provider>
-    );
-    expect(wrapper.find("Notification").exists()).toBe(true);
-    // eslint-disable-next-line testing-library/no-node-access
-    expect(wrapper.find("Notification").props().children).toBe(
-      "Uh oh! It broke"
+    renderWithBrowserRouter(
+      <MachineList
+        grouping={null}
+        hiddenColumns={[]}
+        hiddenGroups={[]}
+        searchFilter=""
+        setHiddenGroups={jest.fn()}
+      />,
+      { route: "/machines", state }
     );
+    expect(screen.getByText("Uh oh! It broke")).toBeInTheDocument();
   });
 
   it("can display a collection of errors", () => {
     state.machine.errors = { machine: "Uh oh!", network: "It broke" };
-    const store = mockStore(state);
-    const wrapper = mount(
-      <Provider store={store}>
-        <MemoryRouter
-          initialEntries={[{ pathname: "/machines", key: "testKey" }]}
-        >
-          <CompatRouter>
-            <MachineList
-              grouping={null}
-              hiddenColumns={[]}
-              hiddenGroups={[]}
-              searchFilter=""
-              setHiddenGroups={jest.fn()}
-            />
-          </CompatRouter>
-        </MemoryRouter>
-      </Provider>
-    );
-    expect(wrapper.find("Notification").exists()).toBe(true);
-    // eslint-disable-next-line testing-library/no-node-access
-    expect(wrapper.find("Notification").props().children).toBe(
-      "machine: Uh oh! network: It broke"
+    renderWithBrowserRouter(
+      <MachineList
+        grouping={null}
+        hiddenColumns={[]}
+        hiddenGroups={[]}
+        searchFilter=""
+        setHiddenGroups={jest.fn()}
+      />,
+      { route: "/machines", state }
     );
+    expect(
+      screen.getByText("machine: Uh oh! network: It broke")
+    ).toBeInTheDocument();
   });
 
   it("dispatches action to clean up machine state when dismissing errors", () => {
     state.machine.errors = "Everything is broken.";
     const store = mockStore(state);
-    const wrapper = mount(
-      <Provider store={store}>
-        <MemoryRouter
-          initialEntries={[{ pathname: "/machines", key: "testKey" }]}
-        >
-          <CompatRouter>
-            <MachineList
-              grouping={null}
-              hiddenColumns={[]}
-              hiddenGroups={[]}
-              searchFilter=""
-              setHiddenGroups={jest.fn()}
-            />
-          </CompatRouter>
-        </MemoryRouter>
-      </Provider>
+    renderWithBrowserRouter(
+      <MachineList
+        grouping={null}
+        hiddenColumns={[]}
+        hiddenGroups={[]}
+        searchFilter=""
+        setHiddenGroups={jest.fn()}
+      />,
+      { route: "/machines", store }
     );
-    wrapper.find("Notification button").simulate("click");
+
+    // Using fireEvent instead of userEvent here,
```

Comment:
This can be fixed by making sure userEvent is advancing fake timers, which you can do by setting up custom userEvent:

```tsx
import {
  userEvent as coreUserEvent,
} from "testing/utils";
  const userEvent = coreUserEvent.setup({ advanceTimers: jest.advanceTimersByTime });
  ```

Reply:
I pushed a commit with this added but it didn't seem to work, I'm getting the same errors as before: https://github.com/canonical/maas-ui/actions/runs/5004143562/jobs/8966334550?pr=4905#step:6:4733

Reply:
strange, let's make sure we investigate this in a followup task https://warthogs.atlassian.net/browse/MAASENG-1712